hvac/v1/__init__.py

```diff
@@ -799,49 +799,49 @@ def create_vault_ec2_client_configuration(self, access_key, secret_key, endpoint
 
-    def get_vault_ec2_client_configuration(self):
+    def get_vault_ec2_client_configuration(self, mount_point='aws-ec2'):
         """
-        GET /auth/aws-ec2/config/client
+        GET /auth/<mount_point>/config/client
         """
-        return self._get('/v1/auth/aws-ec2/config/client').json()
+        return self._get('/v1/auth/{0}/config/client'.format(mount_point)).json()
 
-    def delete_vault_ec2_client_configuration(self):
+    def delete_vault_ec2_client_configuration(self, mount_point='aws-ec2'):
         """
-        DELETE /auth/aws-ec2/config/client
+        DELETE /auth/<mount_point>/config/client
         """
-        return self._delete('/v1/auth/aws-ec2/config/client')
+        return self._delete('/v1/auth/{0}/config/client'.format(mount_point))
 
-    def create_vault_ec2_certificate_configuration(self, cert_name, aws_public_cert):
+    def create_vault_ec2_certificate_configuration(self, cert_name, aws_public_cert, mount_point='aws-ec2'):
         """
-        POST /auth/aws-ec2/config/certificate/<cert_name>
+        POST /auth/<mount_point>/config/certificate/<cert_name>
         """
         params = {
             'cert_name': cert_name,
             'aws_public_cert': aws_public_cert
         }
-        return self._post('/v1/auth/aws-ec2/config/certificate/{0}'.format(cert_name), json=params)
+        return self._post('/v1/auth/{0}/config/certificate/{1}'.format(mount_point, cert_name), json=params)
 
-    def get_vault_ec2_certificate_configuration(self, cert_name):
+    def get_vault_ec2_certificate_configuration(self, cert_name, mount_point='aws-ec2'):
         """
-        GET /auth/aws-ec2/config/certificate/<cert_name>
+        GET /auth/<mount_point>/config/certificate/<cert_name>
         """
-        return self._get('/v1/auth/aws-ec2/config/certificate/{0}'.format(cert_name)).json()
+        return self._get('/v1/auth/{0}/config/certificate/{1}'.format(mount_point, cert_name)).json()
 
-    def list_vault_ec2_certificate_configurations(self):
+    def list_vault_ec2_certificate_configurations(self, mount_point='aws-ec2'):
         """
-        GET /auth/aws-ec2/config/certificates?list=true
+        GET /auth/<mount_point>/config/certificates?list=true
         """
         params = {'list': True}
-        return self._get('/v1/auth/aws-ec2/config/certificates', params=params).json()
+        return self._get('/v1/auth/{0}/config/certificates'.format(mount_point), params=params).json()
 
     def create_ec2_role(self, role, bound_ami_id=None, bound_account_id=None, bound_iam_role_arn=None,
                         bound_iam_instance_profile_arn=None, bound_region=None, bound_vpc_id=None, bound_subnet_id=None,
                         role_tag=None, max_ttl=None, policies=None, allow_instance_migration=False,
-                        disallow_reauthentication=False, **kwargs):
+                        disallow_reauthentication=False, mount_point='aws-ec2', **kwargs):
```

Comment:
note: worth considering how this could negatively callers of this function that took advantage of the "kwargs" bit of this method. Perhaps we should also fill in the rest of the corresponding API routes parameters as explicit keyword arguments to decrease the risk?

Reply:
I'm going to push up a commit to add in those parameters. However after giving it a go, I realize we'll potentially break usage one way or another regardless. E.g., positional arguments could end up as arguments for the wrong parameter. I'll just make sure to call out that fact in the changelog.

Reply:
Actually I just realized I had forgotten about the actual content of https://github.com/ianunruh/hvac/pull/140/ which overlaps here. I'm going to back out 91b263c and see about getting https://github.com/ianunruh/hvac/pull/140/ rebased on top of this mayhaps instead.